resources/run_adf.sh

```diff
@@ -139,72 +177,169 @@ if [ $OS = "Linux" ]; then
+    # Start mongod
     (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_LINK)
 
     # Note: ADF has a storage.json file that generates configs for us.
     # The mongodb source is on port $MONGOD_PORT so we use that here.
     # Uncompress the archive
     if [[ $OS =~ ^CYGWIN ]]; then
-      unzip -o $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE -d $LOCAL_INSTALL_DIR
-
-      # Obtain unzipped directory name
-      MONGO_UNZIP_DIR=$(unzip -l $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE | grep mongod.exe | tr -s ' ' \
-	          | cut -d ' ' -f 5 | cut -d/ -f1)
-      chmod +x $LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR/bin/mongod.exe
-      MONGO_DOWNLOAD_DIR=$LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR
       # mongod does not have --fork option on Windows, using nohup
       nohup $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $(cygpath -m ${MONGO_DB_PATH}) \
               --logpath $(cygpath -m $LOGS_PATH/mongodb_test.log) &
       echo $! > $TMP_DIR/${MONGOD}.pid
     else
-      tar zxvf $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE --directory $LOCAL_INSTALL_DIR
-      MONGO_DOWNLOAD_DIR=$LOCAL_INSTALL_DIR/${MONGO_DOWNLOAD_FILE:0:$((${#MONGO_DOWNLOAD_FILE} - 4))}
       $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH \
         --logpath $LOGS_PATH/mongodb_test.log --pidfilepath $TMP_DIR/${MONGOD}.pid --fork
     fi
+
+    waitCounter=0
+    while : ; do
+        check_mongod
+        if [[ $? -eq 0 ]]; then
+            break
+        fi
+        if [[ "$waitCounter" -gt $TIMEOUT ]]; then
+            echo "ERROR: Local mongod did not start under $TIMEOUT seconds"
+            exit 1
+        fi
+        let waitCounter=waitCounter+1
+        sleep 1
+    done
   fi
 else
   if [ $ARG = $STOP ]; then
     MONGOD_PID=$(< $TMP_DIR/${MONGOD}.pid)
-    echo "Stopping $MONGOD, pid $MONGOD_PID"
+    echo "Stopping data $MONGOD, pid $MONGOD_PID"
     kill "$(< $TMP_DIR/${MONGOD}.pid)"
   fi
 fi
 
-check_mongohoused
-if [[ $? -ne 0 ]]; then
+manage_compute_node() {
   if [ $ARG = $START ]; then
+    check_compute_node $MONGO_DOWNLOAD_DIR
+    if [[ $? -ne 0 ]]; then
+      echo "Starting a compute $MONGOD"
+      mkdir -p ./artifacts/compute-mode-mongod-data
+
+      # Start compute mongod
+      if [[ $OS =~ ^CYGWIN ]]; then
+        # mongod does not have --fork option on Windows, using nohup
+        nohup $MONGO_DOWNLOAD_DIR/bin/mongod -f ./testdata/config/mongod.conf --logpath $(cygpath -m $LOGS_PATH/compute_mongod.log) &
+        echo $! > $TMP_DIR/compute_node.pid
+      else
+        $MONGO_DOWNLOAD_DIR/bin/mongod -f ./testdata/config/mongod.conf --logpath $LOGS_PATH/compute_mongod.log  --pidfilepath $TMP_DIR/compute_node.pid --fork
+      fi
+
+      waitCounter=0
+      while : ; do
+          check_compute_node
+          if [[ $? -eq 0 ]]; then
+              break
+          fi
+          if [[ "$waitCounter" -gt $TIMEOUT ]]; then
+              echo "ERROR: Local mongod did not start under $TIMEOUT seconds"
```

Comment:
```suggestion
              echo "ERROR: Local compute node did not start under $TIMEOUT seconds"
```

I like that you made the distinction in the starting, stopping messages. It would be nice here too, WDYT?